azurerm/resource_arm_iothub.go

```diff
@@ -110,6 +112,105 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"endpoint": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"AzureIotHub.StorageContainer",
+								"AzureIotHub.ServiceBusQueue",
+								"AzureIotHub.ServiceBusTopic",
+								"AzureIotHub.EventHub",
+							}, false),
+						},
+						"connection_string": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateIoTHubEndpointName,
+						},
+						"batch_frequency_in_seconds": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      300,
+							ValidateFunc: validation.IntBetween(60, 720),
+						},
+						"max_chunk_size_in_bytes": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      314572800,
+							ValidateFunc: validation.IntBetween(10485760, 524288000),
+						},
+						"container_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"encoding": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(eventhub.Avro),
+								string(eventhub.AvroDeflate),
+							}, true),
+						},
+						"file_name_format": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateIoTHubFileNameFormat,
+						},
+					},
+				},
+			},
+
+			"route": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(0, 64),
+						},
+						"source": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"DeviceJobLifecycleEvents",
+								"DeviceLifecycleEvents",
+								"DeviceMessages",
+								"Invalid",
+								"TwinChangeEvents",
+							}, false),
+						},
+						"condition": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "true",
```

Comment:
That is intentional - there a special query syntax to filter events before routing them (
https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-query-language#device-to-cloud-message-routes-query-expression). But I think it makes sense to add a comment to avoid confusion in the future.